src/main/java/com/habicus/repository/Loader.java

```diff
@@ -0,0 +1,162 @@
+public class Loader implements ApplicationListener<ApplicationReadyEvent> {
+
+  private static final Logger LOGGER = Logger.getLogger(Loader.class.getName());
+
+  private static final String CONTAINER_PATH = "com.habicus.repository.DataContainers.";
+
+  @Autowired private DataLoaderRegistrar loaderConstants;
+
+  /**
+   * Strips off any dangling file extensions to enable class cast into file ingestor
+   *
+   * @param fileName
+   * @return
+   */
+  private static String normalizeFileName(String fileName) {
+    return fileName.split(".xml")[0];
+  }
+
+  /**
+   * Takes input file from disk and parsed out contents by marshaling XML -> POJO
+   *
+   * @param inputFile
+   * @param classReference
+   * @return
+   * @throws JAXBException
+   */
+  private Container parseFile(Resource inputFile, Class<?> classReference) {
+    try {
+      JAXBContext jaxbContext = JAXBContext.newInstance(classReference);
+      Unmarshaller jaxbUnmarshaller = jaxbContext.createUnmarshaller();
+      return (Container) jaxbUnmarshaller.unmarshal(inputFile.getFile());
+
+    } catch (IOException | JAXBException exception) {
+      LOGGER.log(Level.SEVERE, "Unable to unmarshal data!");
+      exception.printStackTrace();
+    }
+    return null;
+  }
+
+  /**
+   * Read in file from static resources directory
+   *
+   * @param fileResource
+   * @return
+   * @throws IOException
+   * @throws JAXBException
+   * @throws ClassNotFoundException
+   */
+  private Container ingestFromFile(Resource fileResource) {
+    if (fileResource == null) {
+      throw new NullPointerException("File Resource Is Invalid");
+    }
+
+    try {
+      return parseFile(
+          fileResource, forName(CONTAINER_PATH + normalizeFileName(fileResource.getFilename())));
+    } catch (ClassNotFoundException e) {
+      e.printStackTrace();
+    }
+    return null;
+  }
+
+  @Override
+  public void onApplicationEvent(ApplicationReadyEvent event) {
+    LOGGER.log(Level.INFO, "Preparing to load data into database");
+
+    try {
+      loadTestContainers();
+    } catch (IOException | JAXBException e) {
+      LOGGER.log(Level.INFO, "Unable to store user data");
+      e.printStackTrace();
+    }
+  }
+
+  /**
+   * Loads up any test data from resources and stores into embedded DB for test env
+   *
+   * @throws IOException
+   * @throws JAXBException
+   */
+  private void loadTestContainers() throws IOException, JAXBException {
+    LOGGER.log(Level.INFO, "User data table being added to database");
+    Resource[] resources = retrieveTestDataFiles();
+
+    /**
+     * Main Persistence Logic:
+     *
+     * <p>Retrieve data from each file in the resources directory and put into a list Iterate
+     * through each item of the list, cast it to a Container object The getAll operation will
+     * retrieve all sub-elements associated with this data type Put all subelements into a list and
+     * save each individual element into the database
+     */
+    Arrays.stream(resources)
```

Comment:
This is....interesting...

Reply:
@clmaddox Reduced complexity, check it out. https://github.com/Habicus/Habicus-Core/pull/16/commits/453b75a8843d3dd29b003fa941182dac82f3eeb7